### DISCUSSION TITLE
Jul 19 Office Hours: promises in terms, marshalling w/board ids, ...

### DISCUSSION THREAD

**Discussion by dckc** [Agoric/agoric-sdk#8069] [Office Hours]
-  [zoom recording](https://agoric.zoom.us/rec/share/wU2__kCb0mGqux6-VnVthGdAovO-DmwLgvNhGxJssm2jePbMF87YGU7QHRKdT7fF.bEkR0qE6vNe1r46I)  
  Passcode: D1$7hD*c

### 00:00 Introduction

 - [Agoric - Deploy Secure DeFi in JavaScript](https://agoric.com/)
 - [Office Hours · Agoric/agoric-sdk Wiki](https://github.com/Agoric/agoric-sdk/wiki/Office-Hours)

### 00:20 Community update

 - [Liquid Staking Token Vaults \(stATOM, stkATOM, qATOM, etc\.\) \- Inter Protocol / Discussion \- Agoric Community Forum](https://community.agoric.com/t/liquid-staking-token-vaults-statom-stkatom-qatom-etc/346)
 - [Parameter updates: ATOM onboarding and mint limits \- Inter Protocol / Econ Committee \- Agoric Community Forum](https://community.agoric.com/t/parameter-updates-atom-onboarding-and-mint-limits/355/2)
 - [IST Analytics Data \- Should Vault Data be Suppressed? \- Inter Protocol / Discussion \- Agoric Community Forum](https://community.agoric.com/t/ist-analytics-data-should-vault-data-be-suppressed/387)

### 07:00 Cross-chain dapps using async JavaScript

 - [Joe Schnetzler on Twitter: "A beautiful thing @DeanTribble brought up today about @agoric that I had become so accustomed to is the ease of asynchronous to synchronous code with IBC. Send a packet, get a promise, await for ack, continue action. On the Golang SDK side, this is a much more brutal process." / Twitter](https://twitter.com/SchnetzlerJoe/status/1679683985801195520)
 - [JavaScript Framework for Secure Distributed Computing | Documentation](https://docs.agoric.com/guides/js-programming/)

### 09:00 Promises in Contract Terms?

 - [Error: value is not durable: (an object) at slot 7 of (a string)](https://gist.github.com/Jorge-Lopes/391ee1ca57b2f3d1ce62004056ea51a1)
 - [agoric-sdk/packages/inter-protocol/src/proposals/econ-behaviors.js at 6f740a40710dd9f1cff8ea8dddc88cf9d6601486 · Agoric/agoric-sdk](https://github.com/Agoric/agoric-sdk/blob/6f740a40710dd9f1cff8ea8dddc88cf9d6601486/packages/inter-protocol/src/proposals/econ-behaviors.js#L116C14-L116C31) 

p.s.
 - [Zoe can't put terms containing promises in durable storage · Agoric/agoric\-sdk · Discussion \#8073](https://github.com/Agoric/agoric-sdk/discussions/8073)

### 16:09 `makeRecorderKit` vs `makeERecorderKit`

 - [`makeRecorderKit` vs `makeERecorderKit` · Agoric/agoric-sdk · Discussion #8065](https://github.com/Agoric/agoric-sdk/discussions/8065)

### 18:48 Board Marshaling Conventions

 - [0-smart-wallet-vstorate-board-marshal.md](https://gist.github.com/dckc/d333754415b278fc8c9f6a76b6adbb60)
 - [agoric-sdk/packages/inter-protocol/test/auction/snapshots/test-auctionContract.js.md at master · Agoric/agoric-sdk](https://github.com/Agoric/agoric-sdk/blob/master/packages/inter-protocol/test/auction/snapshots/test-auctionContract.js.md)
 - [Eventual Send with E\(\) \| Documentation](https://docs.agoric.com/guides/js-programming/eventual-send.html#eventual-send)
 - [dapp-card-store/contract/deploy.js](https://github.com/Agoric/dapp-card-store/blob/main/contract/deploy.js)
 - [continuing invitation pattern · Issue #667 · Agoric/documentation](https://github.com/Agoric/documentation/issues/667)

### 1:15:55 beta dapp architecture vs mainnet-1

 - [Ag Solo wallet recovery plan · Issue #2629 · Agoric/agoric-sdk](https://github.com/Agoric/agoric-sdk/issues/2629)
 - [How does Smart-Wallet work? · Agoric/agoric-sdk · Discussion #6934](https://github.com/Agoric/agoric-sdk/discussions/6934)

### 1:20:15 funding an account

 - [agoric\-sdk/packages/cosmic\-swingset/Makefile at master · Agoric/agoric\-sdk](https://github.com/Agoric/agoric-sdk/blob/master/packages/cosmic-swingset/Makefile)

**Comment by gacogo** [comment]
Shoutout @dckc  for this session! The  walkthrough on marshalling and the cap data structure is an excellent starting point for anyone exploring these topics. It’s especially helpful for debugging. Highly recommend
